server_test.go

```diff
@@ -167,6 +198,153 @@ func TestLocalServer(t *testing.T) {
+	// the global server has actually responded, unless we introduce a small timeout
+	GlobalResponseChan := make(chan struct{})
+	defer func() {
+		select {
+		case <-GlobalResponseChan:
+			// all is safe
+		case <-time.After(DefaultServerTimeout):
+			assert.Fail(t, "Global server did not complete all responses before test terminated!")
+		}
+	}()
+
+	// We can't be guaranteed that we will receive this exact stream
+	// but we should expect that they represent the same underlying tdigest
+	const ExpectedGobStream = "\r\xff\x87\x02\x01\x02\xff\x88\x00\x01\xff\x84\x00\x007\xff\x83\x03\x01\x01\bCentroid\x01\xff\x84\x00\x01\x03\x01\x04Mean\x01\b\x00\x01\x06Weight\x01\b\x00\x01\aSamples\x01\xff\x86\x00\x00\x00\x17\xff\x85\x02\x01\x01\t[]float64\x01\xff\x86\x00\x01\b\x00\x00/\xff\x88\x00\x05\x01\xfe\xf0?\x01\xfe\xf0?\x00\x01@\x01\xfe\xf0?\x00\x01\xfe\x1c@\x01\xfe\xf0?\x00\x01\xfe @\x01\xfe\xf0?\x00\x01\xfeY@\x01\xfe\xf0?\x00\x05\b\x00\xfeY@\x05\b\x00\xfe\xf0?\x05\b\x00\xfeY@"
+
+	var HistogramValues = []float64{1.0, 2.0, 7.0, 8.0, 100.0}
+	const ABCCountRaw = 5
+	const ABCCountNormalized = 100
+
+	const XYZCountSize = 40
+	expectedMetrics := map[string]float64{
+		// 40 events/50ms = 800 events/s
+		"x.y.z":     800,
+		"a.b.c.max": 100,
+		"a.b.c.min": 1,
+
+		// Count is normalized by second
+		// so 5 values/50ms = 100 values/s
+		"a.b.c.count": ABCCountNormalized,
+	}
+
+	// Set up a remote server (the API that we're sending the data to)
+	// (e.g. Datadog)
+	remoteServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		zr, err := zlib.NewReader(r.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var ddmetrics DDMetricsRequest
+
+		err = json.NewDecoder(zr).Decode(&ddmetrics)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, len(expectedMetrics), len(ddmetrics.Series), "incorrect number of elements in the flushed series on the remote server")
+		assertMetrics(t, ddmetrics, expectedMetrics)
+
+		RemoteResponses++
+		w.WriteHeader(http.StatusAccepted)
+	}))
+
+	// This represents the global veneur instance, which receives request from
+	// the local veneur instances, aggregates the data, and sends it to the remote API
+	// (e.g. Datadog)
+	globalVeneur := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		zr, err := zlib.NewReader(r.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		type requestItem struct {
+			Name      string      `json:"name"`
+			Tags      interface{} `json:"tags"`
+			Tagstring string      `json:"tagstring"`
+			Type      string      `json:"type"`
+			Value     []byte      `json:"value"`
+		}
+
+		var metrics []requestItem
+
+		err = json.NewDecoder(zr).Decode(&metrics)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		assert.Equal(t, 1, len(metrics), "incorrect number of elements in the flushed series")
+
+		tdExpected := tdigest.NewMerging(100, false)
+		err = tdExpected.GobDecode([]byte(ExpectedGobStream))
```

Comment:
question: if we aren't depending on this gob to be exactly equal to the gob that we receive, then do we really need it? eg instead of decoding this t-digest from a gob, we could just create a new t-digest and insert the same samples (ie the `HistogramValues` slice) as the ones we expect.